frontend/src/components/core/StreamlitDialog/ThemeCreator.test.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license
+ * Copyright 2018-2021 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react"
+import { CustomThemeConfig } from "autogen/proto"
+import { mount, shallow } from "lib/test_util"
+import ColorPicker from "components/shared/ColorPicker"
+import UISelectbox from "components/shared/Dropdown"
+import { darkTheme, toThemeInput } from "theme"
+import ThemeCreator, { Props } from "./ThemeCreator"
+
+const getProps = (extend?: Partial<Props>): Props => ({
+  themeInput: toThemeInput(darkTheme.emotion),
+  updateThemeInput: jest.fn(),
+  ...extend,
+})
+
+Object.assign(navigator, {
+  clipboard: {
+    writeText: jest.fn(),
+  },
+})
+
+window.HTMLElement.prototype.scrollIntoView = jest.fn()
+
+describe("Renders ThemeCreator", () => {
+  const props = getProps()
+  const wrapper = shallow(<ThemeCreator {...props} />)
+
+  it("renders closed theme creator crashing", () => {
+    expect(wrapper).toMatchSnapshot()
+  })
+
+  it("Renders opened theme creator", () => {
+    wrapper.find("Button").simulate("click")
+    expect(wrapper).toMatchSnapshot()
+  })
+})
+
+describe("Opened ThemeCreator", () => {
+  const props = getProps()
+  const wrapper = mount(<ThemeCreator {...props} />)
+
+  beforeEach(() => {
+    wrapper.find("Button").simulate("click")
+  })
```

Comment:
Should a new `ThemeCreator` `wrapper` be created in this `beforeEach` too? I think that for every test after the first one, the button click that's being clicked on is the copy to clipboard button instead of the edit theme button